flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/CatalogFactory.java

```diff
@@ -38,20 +37,7 @@
  * instead.
  */
 @PublicEvolving
-public interface CatalogFactory extends TableFactory, Factory {
-
-    /**
-     * Creates and configures a {@link Catalog} using the given properties.
-     *
-     * @param properties normalized properties describing an external catalog.
-     * @return the configured catalog.
-     * @deprecated Use {@link this#createCatalog(Context)} instead and implement {@link Factory}
-     *     instead of {@link TableFactory}.
-     */
-    @Deprecated
-    default Catalog createCatalog(String name, Map<String, String> properties) {
```

Comment:
Due to we have removed this deprecated method, should we change the method 
```
default Catalog createCatalog(Context context) {
        throw new CatalogException("Catalog factories must implement createCatalog(Context)");
    }```
to 

```Catalog createCatalog(Context context);
```

The Catalog implementor must implement this method, so I think give it a default implement is not required. cc@xuyangzhong

Reply:
Since these methods already exist in the `Factory` interface, I directly removed them from `CatalogFactory`.

Reply:
Excuse me, although I think it's better to delete it directly, it would be best to discuss it in the email to inform more people. What do you think? @hackergin @lsyldliu

Reply:
https://lists.apache.org/thread/tlk64byvxtbkox3wyv2jfoqfdtf523bf